src/class.rs

```diff
@@ -169,11 +222,23 @@ impl<T: DfuRuntimeOps, B: UsbBus> UsbClass<B> for DfuRuntimeClass<T> {
             DFU_REQ_DETACH => {
                 self.timeout = self.ops.allow(req.value);
                 if self.timeout.is_some() {
+                    self.state = DfuState::AppDetach;
                     xfer.accept().ok();
                 } else {
+                    self.status = DfuStatusCode::ErrStalledPkt;
+                    self.state = DfuState::DfuError;
                     xfer.reject().ok();
                 }
             },
+            DFU_REQ_CLRSTATUS => {
+                self.status = DfuStatusCode::OK;
+                if self.timeout == None {
+                    self.state = DfuState::AppIdle;
+                } else {
+                    self.state = DfuState::AppDetach;
+                }
+                xfer.accept().ok();
+            },
             _ => { xfer.reject().ok(); },
```

Comment:
Shouldn't we add `self.status = DfuStatusCode::ErrStalledPkt;` here to be consistent?